diffsims/libraries/structure_library.py

```diff
@@ -56,6 +56,36 @@ def __init__(self,
         for ident, struct, ori in zip(identifiers, structures, orientations):
             self.struct_lib[ident] = (struct, ori)
 
+    def get_library_size(cls, to_print = False):
+        """
+        Returns the the total number of orientations in the
+        current StructureLibrary object. Will also print the number of orientations
+        for each identifier in the library if the parameter to_print is set to
+        True. Works with both cls.orientations on the form [1,2]
+        and [[1,2], [2,3]](rotation lists).
+
+        Parameters
+        ----------
+        to_print : bool
+            Default is 'False'
+            Returns
+            -------
+        size_library : int
+            Total number of entries in the current StructureLibrary object.
+        """
+        size_library = 0
+        for i in range (len(cls.orientations)):
+            if type(cls.orientations[i]) != list:
```

Comment:
Can you explain the case you think this catches?

Reply:
So, in some cases like in "test_from_orientations_method" we have :
`orientations = [3, 4]`
which gives 
`type(self.orientations[i]) != tuple `(list is wrong, as you pointed out earlier)

in the above case, we know that each identifier has a unique orientation, and size_library can be incremented by a flat 1.

In the case where we use a rotation_list for each identifier we have:
`orientations = [(3,4), (4,4,6,7,9)] `
which gives
`type(self.orientations[i]) == tuple `

In the latter case size_library must be incremented by the number of elements in the tuple corresponding to each identifier.

Reply:
I think this is wrong, but I also think you've figured that out yourself by the looks of your remarks below?

Reply:
Now that I have changed rotation list to the form of [[(0,0,0),(0,1,0)],[(0,1,0),(0,2,0)]], I have altered this line of code to check
`if len(self.orientations[i]) == 1:
                size_library += 1
else:
                size_library += len(self.orientations[i])`